17.md

```diff
@@ -108,6 +108,50 @@ It's advisable that users choose relays that conform to these practices.
 
 Clients SHOULD guide users to keep `kind:10050` lists small (1-3 relays) and SHOULD spread it to as many relays as viable.
 
+## Key Aliases
+
+Key Aliases hide the user's main key from the public gift wrap. 
+
+Key Alias Events use `kind:10059` replaceable events to privately inform peers of the set of pub keys a user is ALSO subscribing their GiftWrap filters with. The event MUST include a `p` tag for each active public key with an optional relay url to be used with that key. These keys work in addition to the user's main public key. The `relay` url MAY or MAY NOT be included in the user's private inbox relay list (i.e. kind `10050`),
+
+```js
+{
+  "pubkey": "<Author's Main PubKey>",
+  "kind": 10059,
+  "created_at": now(),
+  "tags": [
+    ["p", "<Pubkey Alias 1>", "<preferred relay url for alias 1>"]
+    ["p", "<Pubkey Alias 2>", "<preferred relay url for alias 2>"]
+    ["p", "<Pubkey Alias 3>", "<preferred relay url for alias 3>"]
+    ["expiration", "1600000000"] // optional
+  ]
+}
+```
+
+`kind:10059` MUST be unsigned, sealed, and gift-wrapped to each receiver individually. They MAY or MAY NOT contain the same aliases for every receiver.
```

Comment:
I don't like this. The idea of having to know in advance who you're going to receive messages from ahead of receiving the messages doesn't function without first performing part of the conversation using GWs and normal pubkeys.

Reply:
You don't need to know in advance. The base is the raw NIP-17 protocol: just send the message to the pubkey. Once the conversation gets going you can switch to other keys if clients support it. 

The key alias event is very similar to your "Conversation Request" event on #1206. The only difference is that key alias events can happen (or changed) at any point during a regular chat.